Create new slot for calendar day content (#158)
AlaskaAirlines/auro-datepicker: demo/api.md

<!-- AURO-GENERATED-CONTENT:END -->
</auro-accordion>

### Day Slot

Each day within the calendar has a slot that allows a user to add custom content. The user will add their slot content withtin the `auro-datepicker`, with the slot name needing to be in the format of `date_{month}_{day}_{fullYear}`. **Note**: When the month is a single digit, a zero will need to be added to the front of the month. For example, January would be `01` and not `1`.

> In order to style your slot content, we recommend using in-line styles on your component, as the day slot content is passed down multiple levels within the component. Another way to style the component is to target the `part="daySlot"` attribute that is added to the container around the slot content.

Reviewer: ```suggestion
For `<slot>` styling, it is recommended to use in-line styles. 

... WE NEED A HTML EXAMPLE HERE ....

Another option is to target the `daySlot` CSS Part. This will target the HTML node surrounding the slotted content. 

... WE NEED A CSS EXAMPLE HERE ....
```

Author: Unfortunately, an example using css parts cannot be added at this time. Due to the way the slot content is getting passed down to the auro-calendar-cell level on render, adding a css rule within the html for the specific part attribute does not target the attribute correctly, nor apply the style.

That being said, I completely agree with your requested change, I am just not sure how to correctly implement the example at this time.

> **Disclaimer**: At this time, we are only supporting the use of text and `auro-icon` as the content being used in the slot.

Reviewer: ```suggestion
**Disclaimer**: This element only supports the use of default text or Auro icons to be slotted in the calendar at this time. 
```

> In this example, we use text content within the day slot to display prices for each day.

Reviewer: ```suggestion
In the following example please see how pricing content is threaded into the elements `<slot>`. 
```

  <span slot="fromLabel">Choose a date</span>
  <span slot="mobileDateLabel">Choose a date</span>
  <span slot="date_09_01_2023">$132</span>
  <span slot="date_09_12_2023">$1000</span>
  <span slot="date_09_21_2023">$242</span>
  <span slot="date_10_01_2023">$1322</span>
  <span slot="date_10_02_2023">$234</span>
  <span slot="date_10_11_2023">$784</span>
  <span slot="date_10_15_2023">$567</span>
  <span slot="date_10_16_2023">$12345</span>
  <span slot="date_10_17_2023">Sold</span>
  <span slot="date_10_18_2023">$2000</span>
  <span slot="date_10_19_2023">$1500</span>
  <span slot="date_10_20_2023">$870</span>
  <span slot="date_10_21_2023">$45</span>
  <span slot="date_11_01_2023">$1000</span>
  <span slot="date_11_02_2023">$200</span>
  <span slot="date_11_04_2023">$1540</span>
  <span slot="date_11_08_2023">$2320</span>
  <span slot="date_11_18_2023">$100</span>
  <span slot="date_12_01_2023">$23</span>
  <span slot="date_12_06_2023">$5600</span>
  <span slot="date_01_01_2024">$202</span>
  <span slot="date_01_05_2024">$3022</span>
</auro-datepicker>
```
<!-- AURO-GENERATED-CONTENT:END -->
</auro-accordion>

#### Day Slot Range w/ Text

> Day slot content used with `range` support.

Reviewer: ```suggestion
The following example illustrates slotted content in combination with the `range` support. 
```

  <span slot="fromLabel">Choose a date</span>
  <span slot="mobileDateLabel">Choose a date</span>
  <span slot="date_09_01_2023">$132</span>
  <span slot="date_09_12_2023">$1000</span>
  <span slot="date_09_21_2023">$242</span>
  <span slot="date_10_01_2023">$1322</span>
  <span slot="date_10_02_2023">$234</span>
  <span slot="date_10_11_2023">$784</span>
  <span slot="date_10_15_2023">$567</span>
  <span slot="date_10_16_2023">$1234</span>
  <span slot="date_10_17_2023">$888</span>
  <span slot="date_10_18_2023">$2000</span>
  <span slot="date_10_19_2023">$1500</span>
  <span slot="date_10_20_2023">$870</span>
  <span slot="date_10_21_2023">$45</span>
  <span slot="date_11_01_2023">$1000</span>
  <span slot="date_11_02_2023">$200</span>
  <span slot="date_11_04_2023">$1540</span>
  <span slot="date_11_08_2023">$2320</span>
  <span slot="date_11_18_2023">$100</span>
  <span slot="date_12_01_2023">$23</span>
  <span slot="date_12_06_2023">$5600</span>
  <span slot="date_01_01_2024">$202</span>
  <span slot="date_01_05_2024">$3022</span>
</auro-datepicker>
```
<!-- AURO-GENERATED-CONTENT:END -->
</auro-accordion>

#### Day Slot w/ Icon

> In this example, we show `auro-icon` being used as day slot content.

Reviewer: ```suggestion
The following example illustrates how the `<auro-icon>` element can be used as slotted content. 
```
